base/libdl.jl

```diff
@@ -7,8 +7,8 @@ export DL_LOAD_PATH, RTLD_DEEPBIND, RTLD_FIRST, RTLD_GLOBAL, RTLD_LAZY, RTLD_LOC
     dlpath, find_library, dlext, dllist
 
 const DL_LOAD_PATH = ByteString[]
-@osx_only push!(DL_LOAD_PATH, "@executable_path/../lib/julia")
-@osx_only push!(DL_LOAD_PATH, "@executable_path/../lib")
+@osx_only push!(DL_LOAD_PATH, "@loader_path/../lib/julia")
+@osx_only push!(DL_LOAD_PATH, "@loader_path/../lib")
```

Comment:
since `libjulia.dylib` is performing the `dlopen` call (and thus defines the base path), the `../lib` is redundant. just `"@loader_path"` and `"@loader_path/julia` should work even better.